trackintel/geogr/distances.py

```diff
@@ -170,3 +169,96 @@ def meters_to_decimal_degrees(meters, latitude):
         An approximation of a distance (given in meters) in degrees.
     """
     return meters / (111.32 * 1000.0 * cos(latitude * (pi / 180.0)))
+
+
+def check_wgs_for_distance_calculation(crs):
+    """
+    This functions evaluates if the crs is wgs 84 and warns if no crs is defined.
+
+    Parameters
+    ----------
+    crs : 'pyproj.crs.crs.CRS'
+        Can be the result of `gdf.crs`
+
+    Returns
+    -------
+    is_wgs : bool
+        True if wgs84 is assumed
+
+    Notes
+    ______
+    We do not check for planar crs as this is already done when geopandas.length is called.
+
+    Examples
+    ---------
+    >>> from trackintel.geogr.distances import check_wgs_for_distance_calculation
+    >>> check_wgs_for_distance_calculation(triplegs.crs)
+    """
+
+    is_wgs = False
+
+    if crs == 4326:
+        is_wgs = True
+
+    elif crs is None:
+        is_wgs = True
+        warnings.warn('Your data is not projected. WGS84 is assumed and for length calculation the haversine '
+                      'distance is used')
+    return is_wgs
+
+
+def calc_haversine_length_of_linestrings(gdf):
+    """
+    Calculate the length of linestrings using the haversine distance.
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame with linestring geometry
+        The coordinates are expected to be in WGS84
+
+    Returns
+    -------
+    length: Pandas Series
+        The length of each linestring in meters
+
+    Examples
+    --------
+    >>> from trackintel.geogr.distances import calc_haversine_length_of_linestrings
+    >>> triplegs['length'] = calc_haversine_length_of_linestrings(triplegs)
+    """
+
+    assert all(gdf.geom_type == "LineString")
+
+    length = gdf.geometry.apply(calc_haversine_length_of_single_linestring)
+    return length
+
+
+def calc_haversine_length_of_single_linestring(linestring):
```

Comment:
This could be named to `_calc_haversine_length_of_linestring_single` to highlight the functions relation

Reply:
I agree for the naming in general. But I think this function will potentially be used outside this file so I would take your name without the leading underscore